test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckYourAnswersAndSubmitControllerSpec.scala

```diff
@@ -101,25 +102,66 @@ class CheckYourAnswersAndSubmitControllerSpec extends ControllerSpec with AuthSu
     )
   }
 
-  val mrn: MRN                                                                          = sample[MRN]
-  val declarantTypeAnswer: DeclarantTypeAnswer                                          = sample[DeclarantTypeAnswer]
-  val completeClaimantDetailsAsIndividualAnswer: CompleteDetailsRegisteredWithCdsAnswer =
-    sample[CompleteDetailsRegisteredWithCdsAnswer]
-  val basisOfClaim: BasisOfClaim                                                        = sample[BasisOfClaim]
-  val supportingEvidences: SupportingEvidencesAnswer                                    = sample[SupportingEvidencesAnswer]
-  val completeDutiesSelectedAnswer: DutiesSelectedAnswer                                = sample[DutiesSelectedAnswer]
-  val commodityDetailsAnswer: CommodityDetails                                          = sample[CommodityDetails]
-  val completeNorthernIrelandAnswer: CompleteNorthernIrelandAnswer                      = sample[CompleteNorthernIrelandAnswer]
-  val claimsAnswer: ClaimsAnswer                                                        = sample[ClaimsAnswer]
+  val mrn: MRN                                                     = sample[MRN]
+  val declarantTypeAnswer: DeclarantTypeAnswer                     = AssociatedWithRepresentativeCompany
+  val basisOfClaim: BasisOfClaim                                   = sample[BasisOfClaim]
+  val supportingEvidences: SupportingEvidencesAnswer               = sample[SupportingEvidencesAnswer]
+  val completeDutiesSelectedAnswer: DutiesSelectedAnswer           = sample[DutiesSelectedAnswer]
+  val commodityDetailsAnswer: CommodityDetails                     = sample[CommodityDetails]
+  val completeNorthernIrelandAnswer: CompleteNorthernIrelandAnswer = sample[CompleteNorthernIrelandAnswer]
+  val claimsAnswer: ClaimsAnswer                                   = sample[ClaimsAnswer]
+  val signedInUserDetails                                          = sample[SignedInUserDetails]
+  val declarantEstablishmentAddress                                = EstablishmentAddress(
+    addressLine1 = "line-1",
+    addressLine2 = None,
+    addressLine3 = None,
+    postalCode = Some("AB11C22"),
+    countryCode = "GB"
```

Comment:
Would it be better Magnolia to generate that type with random values?

Reply:
Done